services/terrain/src/terrain/core.clj

```diff
@@ -191,6 +198,7 @@
 
 (def secured-routes-handler
   (-> (delayed-handler secured-routes)
+      (wrap-routes wrap-tree-urls-config)
```

Comment:
I'm not completely sure I like this middleware approach, but some of it's complicated by the fact terrain isn't particularly granular. If we were using compojure-api the whole way down, I'm pretty sure we'd just add this in a `:middlewares` key on just the individual routes which need this configuration to be present.